### PR TITLE
feat: Programatic SEO

### LINKED ISSUE #107: Programatic SEO
Create automatic page for every Capacitor plugin is in Capgo org, and list them here: 
https://capgo.app/plugins/
Design should be changed.
Each plugin should have his page at : https://capgo.app/plugins/NAME
Each plugin page should show README info, GitHub stars, NPM download and have link to GitHub and NPM
We can maybe use GPT to generate a basic tutorial for each.

### PR DESCRIPTION
**Pull request by rishi-raj-jain**
fix #107
/claim #107

### CODE REVIEW

**Comment by rishi-raj-jain** [comment]
@riderx 

Can you take a look at https://deploy-preview-108--capgo.netlify.app/plugins/ and let me know if any changes with icons are required?

**Comment by riderx** [comment]
Very nice! Maybe there some to exclude but that cool 👍 
Can you prepare exclude list ?

**Comment by rishi-raj-jain** [comment]
What do you mean by exclude? Where can I gather what to exclude?

**Comment by rishi-raj-jain** [comment]
@riderx

I'm updating the code with a bearer token env (GITHUB_BEARER_TOKEN) that's required to fetch stars & readme from GitHub. Let me know if there's another way to get around this.

**Comment by rishi-raj-jain** [comment]
@riderx This is ready for review. You'd still need to add your env of BEARER_TOKEN to the git repo (or wherever pnpm build gets executed)

**Comment by rishi-raj-jain** [comment]
Here's where I've used the env:
- https://github.com/Cap-go/website/pull/108/files#diff-faff1af3d8ff408964a57b2e475f69a6b7c7b71c9978cccc8f471798caac2c88
- https://github.com/Cap-go/website/pull/108/files#diff-b14d1f40db48a7e636e94195750364025f820ec100e5e00855279d67e210f331R17

**Comment by rishi-raj-jain** [comment]
@riderx 

Can you follow https://stackoverflow.com/questions/64468843/netlify-deployment
to get rid of
Screenshot 2023-08-18 at 11 09 58 PM
-failed-during-stage-building-site-build-script-returned-n to

**Comment by riderx** [comment]
@rishi-raj-jain thanks a lot, can you share what rights i should but in the token ?

**Comment by riderx** [comment]
For the plugin page we need more content we look for SEO here, so we need to use the REAME file to create automatic content. like plugin presentation and tutorial made with gpt for example

**Comment by rishi-raj-jain** [comment]
Hey riderx,

You can create a classic GitHub token, and no need to select any checkbox there, we just need the API key to be able to fetch public repository worldwide.

File: src/config/plugins.ts [hunk 1]
```diff
@@ -0,0 +1,273 @@
...
+  githubStars?: number
+  readme?: string
+}
+
+export const actions = [
+  {
+    name: '@capgo/cli',
+    author: 'github.com/riderx',
+    description: 'A CLI to upload to capgo servers',
+    href: 'https://github.com/Cap-go/capgo-cli',
```
Comment (riderx) [inline src/config/plugins.ts:52]: this list has to be dynamic from npm not static that why i asked for exclude list

**Comment by rishi-raj-jain** [comment]
Hey @riderx,

1. What do you mean by plugin presentation? The OG image card?

2. What do you mean by tutorial? A usage of the package in which environment?

**Comment by riderx** [comment]
i just added the token in github and netlify

**Comment by rishi-raj-jain** [comment]
I think we should create all of that statically instead of doing real time as this'd require both open ai key and GitHub api key, that can rate limit given the number of PRs and build process that kicks in for each commit.

**Comment by rishi-raj-jain** [comment]
> i just added the token in github and netlify

Named BEARER_TOKEN, right?

File: src/config/plugins.ts (hunk 1 above)
Comment (rishi-raj-jain) [reply src/config/plugins.ts:52]: But then how we plan to inject the icons aptly for the each package?

Comment (rishi-raj-jain) [reply src/config/plugins.ts:52]: I can write a script which we can use pre-building to generate that list, but the issue is rate limiting + which icon to choose dynamically.

Comment (riderx) [reply src/config/plugins.ts:52]: make sense let it like that then

**Comment by riderx** [comment]
i thnk it's ok for now like that,
let forgot about gpt, just make sure we do the maz in SEO, like even jsonLD etc

**Comment by riderx** [comment]
and then i accept it :)

**Comment by rishi-raj-jain** [comment]
I mean I'm open to anything, just shared my concerns that I think will be faced in the long term...

**Comment by rishi-raj-jain** [comment]
@riderx This is ready for review.

**Comment by riderx** [comment]
Ok if you want to try gpt I think you can create a script to make it and we can run it separately or in the CI.
For now I believe the CI will not be too much

**Comment by rishi-raj-jain** [comment]
@riderx

For tutorial, I recommend that we train Langchain first with the plugin and it's api and give it the context to generate the tutorial for.

To do that, I can do upload the trained Langchain and use the openai api on server side to server render the dynamic tutorial. Does that sound good?

**Comment by riderx** [comment]
It's doable in CI context or need more ?

**Comment by rishi-raj-jain** [comment]
So, I'm planning to train locally and then use the key in the CI. I just need the OPENAI_KEY available during the build process.

Hope that clears it up.

**Comment by rishi-raj-jain** [comment]
I love how both plugins & their pages turned out!
- https://deploy-preview-108--capgo.netlify.app/plugins/
- https://deploy-preview-108--capgo.netlify.app/plugins/capgo-cli/

**Comment by rishi-raj-jain** [comment]
How I imagine is this:

- We have a script named `generateTutorial.js` that does the work for us. Creates a directory in the src/content with all the plugins tutorials. This'll be done with the help of the uploaded trained model via OpenAI.

- Before the build process, the script is called that does the magic. Here we'll require the OpenAI key.

- During the build process, the code just loads them from the markdown and then renders it with the website.

This'll allow you to not call the API whenever PRs are deployed via Netlify in Pull Requests, and can choose to run that only on the main branch.

**Comment by rishi-raj-jain** [comment]
@riderx 

1. Can you deploy the latest on Netlify? I think the tutorials look awesome now on the plugins pages.

2. I've updated the CI/CD to generate the tutorials, and dynamically get deployed with each build. In case none of them is found/created, the ones that you see right now will be used.

**Comment by rishi-raj-jain** [comment]
@riderx

I believe this is ready for review. Make sure you've `OPENAI_API_KEY` & `BEARER_TOKEN` set in the your CI.

**Comment by rishi-raj-jain** [comment]
@riderx

I believe this is ready for review. Make sure you've OPENAI_API_KEY & BEARER_TOKEN set in the your CI.

**Comment by rishi-raj-jain** [comment]
@riderx 

I think the tutorials look awesome now on the plugins pages.

Note: I've updated the CI/CD to generate the tutorials, and dynamically get deployed with each build. In case of any errors in generation with any of them is found, the ones that you see right now will be used.

I believe this is ready for review. Make sure you've OPENAI_API_KEY & BEARER_TOKEN set in the your CI.

**Comment by riderx** [comment]
I have add OPENAI_API_KEY and restarted the deploy

**Comment by riderx** [comment]
The generation is very good, just the design has issue on the deployed version i see the readme and then the tutorial, maybe let's do tabs to switch between readme and tuto

**Comment by rishi-raj-jain** [comment]
I'm super glad that you love it! On the way to tabs.

**Comment by rishi-raj-jain** [comment]
here's how the screen now looks

**Comment by rishi-raj-jain** [comment]
@riderx

**Comment by riderx** [comment]
Can the tuto be default ?
and keep the layout for both ? with the info on the left ?
CleanShot 2023-08-21 at 23 15 13@2x

**Comment by riderx** [comment]
it seems the text is not finished here: https://deploy-preview-108--capgo.netlify.app/plugins/capgo-cli/

**Comment by rishi-raj-jain** [comment]
@riderx Done with the changes requested.

**Comment by rishi-raj-jain** [comment]
The build fails because the fetch breaks:

Screenshot 2023-08-22 at 2 55 31 AM


Added the code to handle that.

**Comment by rishi-raj-jain** [comment]
@riderx

This is ready for review!

**Comment by rishi-raj-jain** [comment]
The cool thing about the plugin page is everything is server side rendered! Both the tuto and the readme. This'd help in SEO as you expected/wanted to have.

**Pull request merged**